snmp/tests/common.py

```diff
@@ -0,0 +1,197 @@
+        'name': "IAmString"
+    }
+]
+
+FORCED_METRICS = [
+    {
+        'OID': "1.3.6.1.2.1.4.24.6.0",          # Gauge32
+        'name': "IAmAGauge32",
+        'forced_type': 'counter'
+
+    }, {
+        'OID': "1.3.6.1.2.1.4.31.1.1.6.1",      # Counter32
+        'name': "IAmACounter64",
+        'forced_type': 'gauge'
+    }
+]
+INVALID_FORCED_METRICS = [
+    {
+        'OID': "1.3.6.1.2.1.4.24.6.0",          # Gauge32
+        'name': "IAmAGauge32",
+        'forced_type': 'counter'
+
+    }, {
+        'OID': "1.3.6.1.2.1.4.31.1.1.6.1",      # Counter32
+        'name': "IAmACounter64",
+        'forced_type': 'histogram'
+    }
+]
+
+SCALAR_OBJECTS = [
+    {
+        'OID': "1.3.6.1.2.1.7.1.0",
+        'name': "udpDatagrams"
+    }, {
+        'OID': "1.3.6.1.2.1.6.10.0",
+        'name': "tcpInSegs"
+    }, {
+        'MIB': "TCP-MIB",
+        'symbol': "tcpCurrEstab",
+    }
+]
+
+SCALAR_OBJECTS_WITH_TAGS = [
+    {
+        'OID': "1.3.6.1.2.1.7.1.0",
+        'name': "udpDatagrams",
+        'metric_tags': ['udpdgrams', 'UDP']
+    }, {
+        'OID': "1.3.6.1.2.1.6.10.0",
+        'name': "tcpInSegs",
+        'metric_tags': ['tcpinsegs', 'TCP']
+    }, {
+        'MIB': "TCP-MIB",
+        'symbol': "tcpCurrEstab",
+        'metric_tags': ['MIB', 'TCP', 'estab']
+    }
+]
+
+TABULAR_OBJECTS = [{
+    'MIB': "IF-MIB",
+    'table': "ifTable",
+    'symbols': ["ifInOctets", "ifOutOctets"],
+    'metric_tags': [
+        {
+            'tag': "interface",
+            'column': "ifDescr"
+        }, {
+            'tag': "dumbindex",
+            'index': 1
+        }
+    ]
+}]
+
+INVALID_METRICS = [
+    {
+        'MIB': "IF-MIB",
+        'table': "noIdeaWhatIAmDoingHere",
+        'symbols': ["ifInOctets", "ifOutOctets"],
+    }
+]
+
+PLAY_WITH_GET_NEXT_METRICS = [
+    {
+        "OID": "1.3.6.1.2.1.4.31.3.1.3.2",
+        "name": "needFallback"
+    }, {
+        "OID": "1.3.6.1.2.1.4.31.3.1.3.2.1",
+        "name": "noFallbackAndSameResult"
+    }
+]
+
+
+def generate_instance_config(metrics, template=None):
+    template = template if template else SNMP_CONF
+    instance_config = copy.copy(template)
+    instance_config['metrics'] = metrics
+    instance_config['name'] = HOST
+    return instance_config
+
+
+def generate_v3_instance_config(metrics, name=None, user=None,
+                                auth=None, auth_key=None, priv=None, priv_key=None):
+    instance_config = generate_instance_config(metrics, SNMP_V3_CONF)
+
+    if name:
+        instance_config['name'] = name
+    if user:
+        instance_config['user'] = user
+    if auth:
+        instance_config['authProtocol'] = auth
+    if auth_key:
+        instance_config['authKey'] = auth_key
+    if priv:
+        instance_config['privProtocol'] = priv
+    if priv_key:
+        instance_config['privKey'] = priv_key
+
+    return instance_config
+
+
+def wait_for_async(check, aggregator):
+    for i in range(30):
```

Comment:
Can we log a message when we get out of this loop without receiving a service check ?

Reply:
yes